Source/Server/Managers/DiscordManager.cs

```diff
@@ -0,0 +1,154 @@
+        private static Task updatePlayerCount = new(UpdatePlayerCount);
+        private static int _sendToConsoleDelay = 1000;
+        private static int _updatePlayerCountDelay = 60000;
+
+        private static DiscordSocketConfig Config = new() { GatewayIntents = GatewayIntents.Guilds | GatewayIntents.GuildMessages | GatewayIntents.MessageContent };
+
+        public static async void StartDiscordIntegration()
+        {
+            if (Master.serverConfig == null) return;
+
+            _client = new DiscordSocketClient(Config);
+
+            // Subscribing to events
+            _client.Log += LogAsync;
+            _client.MessageReceived += MessageReceivedAsync;
+
+            await _client.LoginAsync(TokenType.Bot, Master.serverConfig.DiscordIntegration.Token);
+            await _client.StartAsync();
+
+            await _client.SetCustomStatusAsync("");
+            sendToConsoleChannel.Start();
+            updatePlayerCount.Start();
+        }
+
+        private static Task LogAsync(LogMessage log)
+        {
+            if (Master.serverConfig != null && !Master.serverConfig.VerboseLogs) return Task.CompletedTask;
+            
+            Logger.Message("[Discord] " + log.ToString()[9..]);
+            return Task.CompletedTask;
+        }
+
+        private static Task MessageReceivedAsync(SocketMessage message)
+        {
+            if (_client == null) return Task.CompletedTask;
+            if (message.Author.Id == _client.CurrentUser.Id) return Task.CompletedTask; // The bot should never respond to itself.
+            if (message.Author.IsWebhook) return Task.CompletedTask; // Same as above if it's from a webhook
+
+            if (message.Channel.Id == Master.serverConfig.DiscordIntegration.ChatChannelId)
+            {
+                ChatManager.BroadcastDiscordMessage(message.Author.GlobalName == null ? message.Author.Username : message.Author.GlobalName, message.CleanContent);
+            }
+            else if (message.Channel.Id == Master.serverConfig.DiscordIntegration.ConsoleChannelId)
+            {
+                ServerCommandManager.ParseServerCommands(message.CleanContent);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public static async void SendMessageToChatChannel(string user, string message)
+        {
+            if (Master.serverConfig == null) return;
+            if (_client == null) return;
+
+            if (_client.GetChannel(Master.serverConfig.DiscordIntegration.ChatChannelId) is SocketTextChannel channel)
+            {
+                if (Master.serverConfig.DiscordIntegration.ChatWebhook == "") await channel.SendMessageAsync($"{user}: {message}");
+                else SendWebhookMessage(user, message);
+            }
+        }
+
+        public static void SendMessageToConsoleChannelBuffer(string message)
+        {
+            _consoleBuffer.Enqueue(message);
+        }
+
+        private static async void SendMessageToConsoleChannel()
```

Comment:
CancellationToken would be better to avoid runaway loops and to be able to allow the task to be stopped easily.

private static async void SendMessageToConsoleChannel(CancellationToken token)
{
    while (!token.IsCancellationRequested)
    {
        // existing logic...
        await Task.Delay(_sendToConsoleDelay, token);
    }
}